pandas/core/ops.py

```diff
@@ -135,6 +135,13 @@ def rfloordiv(left, right):
 
 
 def rmod(left, right):
+    # check if right is a string as % is the string
+    # formatting operation; this is a TypeError
+    # otherwise perform the op
+    if isinstance(right, compat.string_types):
+        raise TypeError("{typ} cannot perform the operation mod".format(
+            typ=type(left).__name__))
```

Comment:
Why is this needed?

Reply:
OK, I agree that the above is not the best error message (as I assume the catching above is only for a better error message?)

Can you add to the message "... with a string" at the end?

Reply:
Can you do this one?

Reply:
This one